src/main/java/org/zalando/fahrschein/NakadiReader.java

```diff
@@ -315,7 +303,11 @@ private void expectField(JsonParser jsonParser, String expectedFieldName) throws
     private void expectToken(JsonParser jsonParser, JsonToken expectedToken) throws IOException {
         final JsonToken token = jsonParser.nextToken();
         if (token == null) {
-            throw new EOFException(Thread.currentThread().isInterrupted() ? "Thread was interrupted" : "Stream was closed");
+            if (Thread.currentThread().isInterrupted()) {
```

Comment:
When the input stream is interrupted its read method will return -1, which will result in a `null` token. The different exception here is just to get nicer messages in logs, the `IOException` will be caught in the main loop and there is another check if the thread was interrupted.